project-koku/koku
# RBAC caching

This change adds a cache (backed by Redis) to cache RBAC policies for a particular user.  The default Redis config is being used for the time being.

**Testing**
1. Setup Koku to use the RBAC service and run a report API endpoint with a non-admin user.  Verify that the rbac service is contacted and repeated API attempts work using the cached response.
2.  Wait 30 seconds for the cache to expire and ensure RBAC service is contacted on the next request
3.  Stop the redis backend and verify that Koku works with no cache, contacting RBAC service on each request.
4.  Start redis service and verify caching resumes


**Test Results**
[koku_rbac_cache_ut.txt](https://github.com/project-koku/koku/files/3060705/koku_rbac_cache_ut.txt)
Same test performed with OpenShift

![image](https://user-images.githubusercontent.com/4523309/55830037-1ef31f00-5ade-11e9-98a4-42b2b883d911.png)
